### PR TITLE
Update http address during cluster connection

### PR DESCRIPTION
**Pull request by yux0**
## What changed?
Update http address during cluster connection

## Why?
Store remote cluster http address during connection

## How did you test it?
local cluster tested

### CODE REVIEW

File: service/frontend/admin_handler.go [hunk 1]
```diff
@@ -1150,7 +1150,7 @@ func (adh *AdminHandler) AddOrUpdateRemoteCluster(
 			HistoryShardCount:        resp.GetHistoryShardCount(),
 			ClusterId:                resp.GetClusterId(),
 			ClusterAddress:           request.GetFrontendAddress(),
-			HttpAddress:              request.GetFrontendHttpAddress(),
```
Comment (bergundy) [inline service/frontend/admin_handler.go:1153]: Why are we taking the frontend grpc address from the request an the http address from the `resp`?

Comment (yux0) [reply service/frontend/admin_handler.go:1153]: we need to know the grpc address to connect two clusters. From my understanding, we don't have http endpoint on admin service

Comment (bergundy) [reply service/frontend/admin_handler.go:1153]: I think we should be consistent here.

Comment (bergundy) [reply service/frontend/admin_handler.go:1153]: We synced offline, this makes sense to me now.

You could argue that the frontend address (gRPC) should be taken from the response too but it's okay the way it is.

**Review by bergundy — APPROVED** [review]

**Pull request merged**